app/services/google/scrape_service.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+module Google
+  class ScrapeService
+    def initialize(keyword_id:, search_service: Google::SearchService.new)
+      @keyword_id = keyword_id
+      @search_service = search_service
+    end
+
+    def call!
+      keyword = Keyword.find_by id: keyword_id
+      raise_keyword_not_found unless keyword
+      @keyword = keyword
+
+      begin
+        html = search_service.search! keyword.content
+      rescue GoogleScraperRuby::Errors::SearchServiceError => error
+        raise_unexpected_error error
+      end
+
+      result = Google::ParseService.new(html).call
```

Comment:
Above we pass the `search_service` and here we use the `Google::ParseService` directly, so we keep the pattern consistency, should we pass the `parse_service` to this as well 🤔

Reply:
ah yeah, I was thinking about it as well but decide to leave it like this. No particular reason. Your point makes sense

Reply:
Fixed in b5840010db8bda70161194af39239f14661508f9